tests/h/views/api/users_test.py

```diff
@@ -5,183 +5,125 @@
-        TrustedUserJSONPresenter.return_value.asdict.assert_called_once_with()
-
-    def test_it_returns_presented_user(
-        self, pyramid_request, context, TrustedUserJSONPresenter
-    ):
-        result = read(context, pyramid_request)
         assert result == TrustedUserJSONPresenter.return_value.asdict.return_value
 
 
-@pytest.mark.usefixtures("client_authority", "user_signup_service", "user_unique_svc")
+@pytest.mark.usefixtures("user_signup_service", "user_unique_service")
 class TestCreate:
-    def test_signs_up_user(self, pyramid_request, user_signup_service, valid_payload):
-        pyramid_request.json_body = valid_payload
-
-        create(pyramid_request)
-
-        user_signup_service.signup.assert_called_once_with(
-            require_activation=False,
-            authority="weylandindustries.com",
-            username="jeremy",
-            email="jeremy@weylandtech.com",
-            display_name="Jeremy Weyland",
-            identities=[{"provider": "provider_a", "provider_unique_id": "abc123"}],
-        )
-
-    def test_it_presents_user(
-        self, pyramid_request, valid_payload, user, TrustedUserJSONPresenter
+    def test_it(
+        self,
+        pyramid_request,
+        user_signup_service,
+        user_unique_service,
+        auth_client,
+        valid_payload,
+        CreateUserAPISchema,
+        TrustedUserJSONPresenter,
     ):
-        pyramid_request.json_body = valid_payload
-        create(pyramid_request)
-
-        TrustedUserJSONPresenter.assert_called_once_with(user)
+        CreateUserAPISchema.return_value.validate.return_value = valid_payload
 
-    def test_it_returns_presented_user(
-        self, pyramid_request, valid_payload, TrustedUserJSONPresenter
-    ):
-        pyramid_request.json_body = valid_payload
         result = create(pyramid_request)
 
-        assert result == TrustedUserJSONPresenter.return_value.asdict()
-
-    def test_it_validates_the_input(
-        self, pyramid_request, valid_payload, CreateUserAPISchema
-    ):
-        create_schema = CreateUserAPISchema.return_value
-        create_schema.validate.return_value = valid_payload
-        pyramid_request.json_body = valid_payload
-
-        create(pyramid_request)
-
-        create_schema.validate.assert_called_once_with(valid_payload)
+        CreateUserAPISchema.assert_called_with()
+        CreateUserAPISchema.return_value.validate.assert_called_once_with(valid_payload)
+        user_unique_service.ensure_unique.assert_called_with(
+            valid_payload, authority=auth_client.authority
+        )
+        user_signup_service.signup.assert_called_once_with(
+            require_activation=False, **valid_payload
+        )
+        user = user_signup_service.signup.return_value
+        TrustedUserJSONPresenter.assert_called_with(user)
+        TrustedUserJSONPresenter.return_value.asdict.assert_called_once_with()
+        assert result == TrustedUserJSONPresenter.return_value.asdict.return_value
```

Comment:
This is a change from the many single tests with one asset per test to many asserts in one test.